_posts/2024-02-05-cde-customization.adoc

```diff
@@ -0,0 +1,434 @@
+
+### Start a new CDE using the custom image
+ 
+The simplest way to start a workspace using a new custom image is to use https://eclipse.dev/che/docs/stable/end-user-guide/url-parameter-for-container-image/[`?image=<image-name>` Eclipse Che URL parameter]. 
+
+For example, to start a CDE using the custom image `quay.io/mloriedo/rails-blog-cde` and cloning rails-sample, the following string can be used in the "Create Workspace" page:
+
+[attributes]
+....
+https://github.com/l0rd/rails-sample&image=quay.io/mloriedo/rails-blog-cde
+....
+
+image::/assets/img/cde-customization/image-url-parameter.png[]
+{nbsp} +
+
+Or this URL can be used to start it on Red Hat Developer Sandbox: 
+
+[attributes]
+....
+https://workspaces.openshift.com/#https://github.com/l0rd/rails-sample&image=quay.io/mloriedo/rails-blog-cde
+....
+
+### Verify the new CDE Pod image
+
+The CDE Pod uses a `rails-blog-cde` container rather than Eclipse Che default universal developer image. To verify it open a Terminal from VisualStudio Code and run kubectl:
+
+[attributes]
+....
+$ kubectl get pods -o jsonpath='{range ..containers[*]}{.image}{"\n"}{end}'
+quay.io/mloriedo/rails-blog-cde:latest
+registry.redhat.io/devspaces/traefik-rhel8@sha256:f10ddd0e836be6d4cce38453025956a5d041b4841039734ab30c6b7bf1fc7e7e
+....
+
+### Run Ruby on Rails from the Terminal
+
+And we can now verify that the `rails-sample` application starts successfully from the VisualStudio Code Terminal:
+
+[attributes]
+....
+$ bundle install && ./bin/rails server
+(...)
+=> Booting Puma
+=> Rails 7.0.8 application starting in development 
+=> Run `bin/rails server --help` for more startup options
+Puma starting in single mode...
+* Puma version: 5.6.8 (ruby 3.1.2-p20) ("Birdie's Version")
+*  Min threads: 5
+*  Max threads: 5
+*  Environment: development
+*          PID: 4119
+* Listening on http://127.0.0.1:3000
+* Listening on http://[::1]:3000
+Use Ctrl-C to stop
+....
+
+### Install `ruby-lsp` Visual Studio Code extension
+
+Finally, the Ruby LSP extension, that failed to activate in STEP1, is now fully working:
+
+image::/assets/img/cde-customization/ruby-lsp-extension-fully-functional.png[]
+{nbsp} +
+Replacing Eclipse Che default image with a custom one allowed to include the right pre-requisites (ruby and rails) to the CDE. With that we are able to run the application and the VisualStudio Code ruby extension. 
+
+There are still a few properties of the CDE that can be improved (like increasing the CPU and memory, specifying the command to run the application, pre-fetch the dependencies etc...). We will cover those in the next session where we will use the Devfile for CDE more advanced configurations.
+
+[TIP]
+====
+In this part the image of the CDE has been replaced using a URL parameter. This can be done using a Devfile too as we will see in next session. An alternative to those approches is to change the image "on the fly" using `kubectl patch` against the DevWorkspace object:
```

Comment:
```suggestion
In this part the image of the CDE has been replaced using a URL parameter. This can be done using a Devfile too as we will see in the next session. An alternative to those approaches is to change the image "on the fly" using `kubectl patch` against the DevWorkspace object:
```